### PR TITLE
[v11] Modify `teleport configure` to gracefully handle `--auth-server`

### PR DESCRIPTION
**Pull request by strideynet**
Emits a warning and downgrades the outputted configuration to v2.

Closes https://github.com/gravitational/teleport/issues/17890 

```
➜  teleport git:(strideynet/partially-revert-config-v3) ✗ build/teleport configure --auth-server my.proxy:443
#
# A Sample Teleport configuration file.
#
# Things to update:
#  1. license.pem: You only need a license from https://dashboard.goteleport.com
#     if you are an Enterprise customer.
#
version: v2
teleport:
  nodename: Noahs-MBP-2.cust.communityfibre.co.uk
  data_dir: /var/lib/teleport
  auth_servers:
  - my.proxy:443
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  proxy_listener_mode: multiplex
ssh_service:
  enabled: "yes"
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  enabled: "yes"
  https_keypairs: []
  acme: {}

---BREAKING CHANGE IMPENDING---
Configuration version 'v3' (introduced in Teleport 11) requires that '--proxy' is used instead of '--auth-server' when connecting a Node to a Proxy rather than directly to the Auth Server.
We have detected that you may have provided a proxy address with '--auth-server' and have downgraded the generated config to 'v2' to ensure continuity of service.
This behaviour will cease in Teleport 12! Begin using '--proxy' when providing the proxy address as soon as possible.

```

This is being merged directly to `branch/v11`, as we will be discontinuing this fallback behaviour in Teleport 12.

### CODE REVIEW

File: lib/config/fileconf.go [hunk 1]
```diff
@@ -220,12 +221,37 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
...
+			// there is a chance that this is a Proxy.
+			// This ensures we do not break people relying on `--auth-server`.
+			addr, err := utils.ParseAddr(flags.AuthServer)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
 
-	if flags.ProxyAddress != "" {
-		g.ProxyServer = flags.ProxyAddress
+			// Check port to determine if this is obviously a auth server
```
Comment (ryanclark) [inline lib/config/fileconf.go:237]:
```suggestion
			// Check port to determine if this is obviously an auth server
```

**Review by ryanclark — APPROVED** [review]

File: lib/config/fileconf.go [hunk 2]
```diff
@@ -220,12 +221,37 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
 		Method:    types.JoinMethod(joinMethod),
 	}
 
-	if flags.AuthServer != "" {
-		g.AuthServer = flags.AuthServer
-	}
+	if flags.Version == defaults.TeleportConfigVersionV3 {
```
Comment (hatched) [inline lib/config/fileconf.go:224]:
```suggestion
	if version == defaults.TeleportConfigVersionV3 {
```
🤷‍♂️

**Review by hatched — APPROVED** [review]

File: lib/config/fileconf.go (hunk 2 above)
Comment (nklaassen) [inline lib/config/fileconf.go:224]: oh man the default flag value is V3. Would be nice for users to have a way say "trust me, I really want a v3 config with auth_server: auth.example.com:4025".

Suggestion: remove the default flag value at the CLI parser level, then in `(*dumpFlags).CheckAndSetDefaults` we should:
1. Let the user set a specific version if they want to, and then trust whatever they give. `teleport configure --version v3 --auth_server auth.example.com:4025` should do exactly what it says.
2. If the user did not set a specific version and set `--auth_server`, default to v2 and print the warning.
3. Else default to v3 (this includes the cases where they set `--proxy` or not)

Comment (strideynet) [reply lib/config/fileconf.go:224]: 1a846d7842

**Review by nklaassen — APPROVED** [review]
Looks good! Thanks for putting this together

**Pull request merged**